code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -370,11 +370,13 @@
 		return attack_hand(user)
 
 /obj/structure/mineral_door/resin/TryToSwitchState(atom/user)
-	if(islarva(user))
-		var/mob/living/carbon/xenomorph/larva/L = user
-		if (L.hivenumber == hivenumber)
-			L.scuttle(src)
-		return
+	if(isxeno(user))
+		var/mob/living/carbon/xenomorph/X = user
+		if (X.hivenumber == hivenumber)
+			if(X.scuttle(src))
+				return // don't toggle door for small xenos
+			return ..() // toggles door for big xenos
+		return // enemy hive doesn't toggle door
```

Comment:
could use early returns here to make this more understandable

Reply:
Early returned for more understandablility and fixed for allies of the hive